File: Basics_with_JavaScript/910 For Loop/More Exersises/04. Grades.js
function vladi(array) {
  let studentsQty = Number(array[0]);
  let gradeSum = 0;
  let topQty = 0;
  let fourQty = 0;
  let threeQty = 0;
  let failQty = 0;
  let average = 0;

  for (let i = 1; i <= studentsQty; i++) {
    let grade = Number(array[i]);
    if (grade >= 5.0) {
      topQty += 1;
    } else if (grade >= 4.0 && grade <= 4.99) {
      fourQty += 1;
    } else if (grade >= 3.0 && grade <= 3.99) {
      threeQty += 1;
    } else if (grade < 3.0) {
      failQty += 1;
    }
    gradeSum += grade;
  }

  average = (gradeSum / studentsQty).toFixed(2);
  topQty = (topQty * 100 / studentsQty).toFixed(2);
  fourQty = (fourQty * 100 / studentsQty).toFixed(2);
  threeQty = (threeQty * 100 / studentsQty).toFixed(2);
  failQty = (failQty *100 / studentsQty).toFixed(2);
  console.log(`Top students: ${topQty}%`);
  console.log(`Between 4.00 and 4.99: ${fourQty}%`);
  console.log(`Between 3.00 and 3.99: ${threeQty}%`);
  console.log(`Fail: ${failQty}%`);
  console.log(`Average: ${average}`);
}

vladi([
  "10",
  "3.00",
  "2.99",
  "5.68",
  "3.01",
  "4",
  "4",
  "6.00",
  "4.50",
  "2.44",
  "5",
]);
vladi(["6", "2", "3", "4", "5", "6", "2.2"]);

File: Basics_with_Python/pythonBasicBook/!PythonBasicCourse/4_For Loop/10_For Loop - Exercise/More Exercise2/04. Grades.py
students_total = int(input())
grade_var = 0
a = 0
b = 0
c = 0
d = 0
for _number in range(1, students_total + 1):
    grade = float(input())
    grade_var += grade
    if 2.00 <= grade <= 2.99:
        a += 1
    elif 3.00 <= grade <= 3.99:
        b += 1
    elif 4.00 <= grade <= 4.99:
        c += 1
    elif grade >= 5.00:
        d += 1
average = grade_var / students_total
d *= 100/students_total
c *= 100/students_total
b *= 100/students_total
a *= 100/students_total
print(f"Top students: {d:.2f}%")
print(f"Between 4.00 and 4.99: {c:.2f}%")
print(f"Between 3.00 and 3.99: {b:.2f}%")
print(f"Fail: {a:.2f}%")
print(f"Average: {average:.2f}")
